For historical reasons, libndbm is actually Berkeley DB.  Take it out.

--- Python-2.2/setup.py	Fri Jan 25 13:00:29 2002
+++ Python-2.2/setup.py	Fri Jan 25 13:00:25 2002
@@ -375,14 +375,7 @@
 
         # The standard Unix dbm module:
         if platform not in ['cygwin']:
-            if (self.compiler.find_library_file(lib_dirs, 'ndbm')):
-                exts.append( Extension('dbm', ['dbmmodule.c'],
-                                       libraries = ['ndbm'] ) )
-            elif self.compiler.find_library_file(lib_dirs, 'db1'):
-                exts.append( Extension('dbm', ['dbmmodule.c'],
-                                       libraries = ['db1'] ) )
-            else:
-                exts.append( Extension('dbm', ['dbmmodule.c']) )
+            exts.append( Extension('dbm', ['dbmmodule.c']) )
 
         # Anthony Baxter's gdbm module.  GNU dbm(3) will require -lgdbm:
         if (self.compiler.find_library_file(lib_dirs, 'gdbm')):
